api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -454,6 +454,16 @@ type HostedClusterSpec struct {
 	// +required
 	Platform PlatformSpec `json:"platform"`
 
+	// KubeAPIServerDNSName specifies a desired DNS name to resolve to the KAS.
+	// When set, the controller will automatically generate a secret with kubeconfig and expose it in the hostedCluster Status.
+	// If it's set or removed day 2, the kubeconfig generated secret will be created, recreated or deleted.
```

Comment:
Can we document how this field relates to configuration.APIServer.ServingCerts.NamedCertificates?